### PR TITLE
Black box testing

### PR DESCRIPTION
**Pull request by Eirik0**
This commit adds black-box testing for Fq and Fr. It also adds as dev-dependencies: `rand_core` and `rand_xorshiftrng`.

### CODE REVIEW

**Comment by Eirik0** [comment]
This PR depends on #15.

**Comment by ebfull** [comment]
Can you rebase on `master` now?

**Comment by Eirik0** [comment]
Done.

File: tests/common.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    XorShiftRng::from_seed([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15])
+}
+
+pub trait MyRandom {
+    fn new_random<T: RngCore>(rng: &mut T) -> Self;
+}
+
+impl MyRandom for Fq {
+    fn new_random<T: RngCore>(rng: &mut T) -> Self {
+        let mut ranmdom_bytes = [0u8; 64];
```
Comment (ebfull) [inline tests/common.rs]: `ranmdom` -> `random`

File: tests/common.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        rng.fill_bytes(&mut ranmdom_bytes);
+        Fq::from_bytes_wide(ranmdom_bytes)
+    }
+}
+
+impl MyRandom for Fr {
+    fn new_random<T: RngCore>(rng: &mut T) -> Self {
+        let mut ranmdom_bytes = [0u8; 64];
```
Comment (ebfull) [inline tests/common.rs]: also typo here

File: tests/common.rs (hunk 1 above)
Comment (Eirik0) [reply tests/common.rs]: Whoops!

**Comment by Eirik0** [comment]
I thought I had done addition, but obviously not. I've made the suggested changes and force pushed.

**Pull request merged**